src/nvim/api/autocmd.c

```diff
@@ -609,10 +609,13 @@ Integer nvim_create_autocmd(uint64_t channel_id, Object event, Dict(create_autoc
 /// NOTE: Only autocommands created via the API have an id.
 /// @param id Integer The id returned by nvim_create_autocmd
 /// @see |nvim_create_autocmd()|
-void nvim_del_autocmd(Integer id)
+void nvim_del_autocmd(Integer id, Error *err)
   FUNC_API_SINCE(9)
 {
-  autocmd_delete_id(id);
+  bool deleted = autocmd_delete_id(id);
+  if (!deleted) {
+    api_set_error(err, kErrorTypeException, "Failed to delete any autocmds");
```

Comment:
`"Failed to delete autocmd"` I think

Reply:
If id=0 then it will delete multiple.

Reply:
shouldn't `id=0` be an error tho? (can use the same error message)

Reply:
`NOTE: Only autocommands created via the API have an id.` So `id=0` is an indicator of missing id. and should be an error in this function here.

Reply:
Cool, that makes things easier and allows a small optimization since I'm assuming ids are unique.